opencontainers/struct.py

```diff
@@ -293,15 +303,43 @@ def add(self, name, value):
         """
         Add a value to an existing attribute, normally when used by a client
         """
+        if value is None:
+            return
         if name not in self.attrs:
             bot.exit("%s is not a valid attribute." % name)
         attr = self.attrs[name]
 
-        # If we already have a value, update it
-        if attr.value and isinstance(attr.value, dict):
-            value.update(attr.value)
-        elif attr.value and isinstance(attr.value, list):
-            value = list(set(attr.value + [value]))
+        valueType = type(value)
+        if attr._is_struct():
+            # Transform value to Struct if not already
+            if not is_struct(valueType):
+                value = attr.attType().load(value)
+            if attr.attType in [StrStruct, IntStruct]:
+                if attr.value is not None:
+                    value = attr.value + value
+        elif attr.attType == list or isinstance(attr.attType, list):
```

Comment:
Why do we need to check attType and the instance type?

Reply:
This is needed because for attributes with `attType=[IntStruct]` the `isinstance()` check is `True` and for attributes with `attType=list` or `attType=dict` as used in opencontainers/image/v1/config.py.